specification/datacollaboration/resource-manager/Microsoft.DataCollaboration/preview/2020-05-04-preview/DataCollaboration.json

```diff
@@ -0,0 +1,5736 @@
+          "uniqueItems": false,
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "sources": {
+          "description": "The list of parameters the script can use as it's data sources",
+          "uniqueItems": false,
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "visibility": {
+          "description": "Script visibility mode",
+          "enum": [
+            "ShowAll",
+            "HiddenContent"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "visibility",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "Workspace": {
+      "description": "A workspace data transfer object.",
+      "required": [
+        "identity"
+      ],
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/DefaultDto"
+        }
+      ],
+      "properties": {
+        "identity": {
+          "$ref": "#/definitions/Identity",
+          "description": "Identity Info on the Workspace"
+        },
+        "properties": {
+          "$ref": "#/definitions/WorkspaceProperties",
+          "description": "Properties on the workspace",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "WorkspaceList": {
+      "description": "List response for get Workspaces.",
+      "required": [
+        "value"
+      ],
+      "type": "object",
+      "properties": {
+        "nextLink": {
+          "description": "The Url of next result page.",
+          "type": "string"
+        },
+        "value": {
+          "description": "Collection of items of type DataTransferObjects.",
+          "uniqueItems": false,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Workspace"
+          }
+        }
+      }
+    },
+    "WorkspaceProperties": {
+      "description": "Workspace property bag.",
+      "type": "object",
+      "properties": {
+        "createdAt": {
+          "format": "date-time",
+          "description": "Time at which the workspace was created.",
+          "type": "string",
+          "readOnly": true
+        },
+        "provisioningState": {
+          "description": "Provisioning state of the Workspace",
+          "enum": [
+            "Succeeded",
+            "Creating",
+            "Deleting",
+            "Moving",
+            "Failed"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "provisioningState",
+            "modelAsString": true
+          }
+        },
+        "userEmail": {
+          "description": "Email of the user who created the resource",
+          "type": "string",
+          "readOnly": true
+        },
+        "userName": {
+          "description": "Name of the user who created the resource",
+          "type": "string",
+          "readOnly": true
```

Comment:
I would remove these and just rely on systemData

Reply:
We replaced specific properties with system data in all relevant resources